lib/matplotlib/tests/test_triangulation.py

```diff
@@ -1299,3 +1299,14 @@ def test_triplot_with_ls(fig_test, fig_ref):
     data = [[0, 1, 2]]
     fig_test.subplots().triplot(x, y, data, ls='--')
     fig_ref.subplots().triplot(x, y, data, linestyle='--')
+
+
+@check_figures_equal()
+def test_triplot_label(fig_test, fig_ref):
+    x = [0, 2, 1]
+    y = [0, 0, 1]
+    data = [[0, 1, 2]]
+    fig_test.subplots().triplot(x, y, data, label='label')
+    fig_test.axes[0].legend()
+    fig_ref.subplots().triplot(x, y, data)
+    fig_ref.axes[0].legend(['label'])
```

Comment:
I think an explicit logic test is better than an image comparision here:
```suggestion
def test_triplot_label(fig_test, fig_ref):
    x = [0, 2, 1]
    y = [0, 0, 1]
    data = [[0, 1, 2]]
    fig, ax = plt.subplots()
    lines, markers = ax.triplot(x, y, data, label='label')
    handles, labels = ax.get_legend_handles_labels()
    assert labels = ['label']
    assert len(handles) == 1
    assert handles[0] is lines
```